src/bin/sync-categories.rs

```diff
@@ -0,0 +1,48 @@
+// Sync available crate categories from `src/categories.txt`.
+// Only needs to be run for new databases or when `src/categories.txt`
+// has been updated.
+//
+// Usage:
+//      cargo run --bin sync-categories
+
+#![deny(warnings)]
+
+extern crate cargo_registry;
+extern crate postgres;
+
+use cargo_registry::env;
+
+fn main() {
+    let conn = postgres::Connection::connect(&env("DATABASE_URL")[..],
+                                             postgres::TlsMode::None).unwrap();
+    let tx = conn.transaction().unwrap();
+    sync(&tx).unwrap();
+    tx.set_commit();
+    tx.finish().unwrap();
+}
+
+fn sync(tx: &postgres::transaction::Transaction) -> postgres::Result<()> {
+    let categories = include_str!("../categories.txt");
+    let categories: Vec<_> = categories.lines().collect();
+    let insert = categories.iter()
+                           .map(|c| format!("('{}')", c))
+                           .collect::<Vec<_>>()
+                           .join(",");
+    let in_clause = categories.iter()
+                              .map(|c| format!("'{}'", c))
+                              .collect::<Vec<_>>()
+                              .join(",");
```

Comment:
rust-postgres doesn't support this for the bulk import of multiple values that I wanted to do here, see sfackler/rust-postgres#218. I can change the `WHERE NOT IN` to use rust-postgres' escaping, though.

Another option is to make one insert for every category... this doesn't really NEED to be fast... wdyt?